api/sessions_api.go

```diff
@@ -414,40 +228,56 @@ func (s *Session) terminate() {
 	s.SessionManager.DeleteSession(s.ID)
 }
 
-func (s *Session) handleAbnormalClosure() int {
-	log.Printf("starting grace period for %s\n", s.ID)
+// Writes to the connection of that session. It also
+// handles the abnormal or other types of errors of
+// writing to a websocket connection.
+func (s *Session) writeToConn(p interface{}) int {
```

Comment:
can we use communication channel in session manager to send data back to the client? and abstract this function in session manager?

Reply:
Also you could argue that `WriteJSONWithRetry` belongs to `Connection` class and not the session. So you could write that function on `Conn` and call it like this:
```go
s.Conn.WriteJSONWithRetry(p)
```

Reply:
I thought about this.
The things is, session has a pointer to its connection. It's like it's right there to be written to and it makes sense if each session is taking care of itself.
If we give that responsibility to `SessionManager`, it would be another layer of abstraction but I don't know for what benefit.

Wait lemme ask this. why do you think it should be done by `SessionManager` and not `Session` itself?

Reply:
So you want to create a Connection struct separately?

Reply:
ohhh nvm, I thought `Conn` is a struct that we made. Nvm then. it's all good